src/main/java/fr/insee/pogues/utils/CodesListConverter.java

```diff
@@ -0,0 +1,49 @@
+package fr.insee.pogues.utils;
+
+
+import fr.insee.pogues.model.*;
+import fr.insee.pogues.webservice.model.dtd.codeList.Code;
+import fr.insee.pogues.webservice.model.dtd.codeList.CodesList;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CodesListConverter {
+
+    private CodesListConverter() {}
+
+    private static List<CodeType> convertFromCodeListDTDToCodeTypeWithParent(Code code){
+        List<CodeType> codeTypes = new ArrayList<>();
+        CodeType codeType = new CodeType();
+        codeType.setValue(code.getValue());
+        codeType.setLabel(code.getLabel());
+        codeTypes.add(codeType);
+        if(code.getCodes() != null){
+            List<CodeType> subCodeTypes = code.getCodes().stream().map(subCode -> {
+                CodeType subCodeType = new CodeType();
+                subCodeType.setValue(subCode.getValue());
+                subCodeType.setLabel(subCode.getLabel());
+                subCodeType.setParent(code.getValue());
+                return subCodeType;
+            }).toList();
+            codeTypes.addAll(subCodeTypes);
+        }
+        return codeTypes;
+    }
+
+    public static CodeList convertFromCodeListDTDtoCodeListModel(CodesList codesListDtd){
+        CodeList codeList = new CodeList();
+        codeList.setId(codesListDtd.getId());
+        codeList.setLabel(codesListDtd.getLabel());
+        List<CodeType> codeTypes = new ArrayList<>();
+        codesListDtd.getCodes().stream().forEach(code ->
+            codeTypes.addAll(convertFromCodeListDTDToCodeTypeWithParent(code))
+        );
```

Comment:
Normalement tu peux directement faire un truc du style 
```suggestion
        List<CodeType> codeTypes = codesListDtd.getCodes().stream()
            .map(code -> convertFromCodeListDTDToCodeTypeWithParent(code))
            .flatmap(Collection::stream).toList();
```
Le foreach est à éviter quand on peut car on sait pa trop ce qu'on fait dedans (bon là c'est simple ok). C'est une bonne pratique d'avoir des fonctions qui return ce qu'ils font plutot que d'alimenter une var global dans un foreach.
Tu maitrises mieux le code avec ce pattern je trouve (surtout quand les manipulations deviennent plus compliquées)